game/src/test/java/com/hiddenswitch/spellsource/CustomCardsTests.java

```diff
@@ -2220,6 +2221,114 @@ public void testStablePortal() {
 		});
 	}
 
+	@Test
+	public void testCryWolf() {
+		runGym((context, player, opponent) -> {
+			playCard(context, player, "spell_cry_wolf");
+			playCard(context, player, player.getHand().peekFirst());
+			playCard(context, player, player.getHand().peekFirst());
+
+			print(player.getMinions().get(0).getSourceCard().getCardId());
+			print(player.getMinions().get(1).getSourceCard().getCardId());
+			print(player.getMinions().get(2).getSourceCard().getCardId());
+		});
+
+	}
+
+	@Test
+	public void testGnarlRoot() {
+		runGym((context, player, opponent) -> {
+			for (int i = 0; i < 9; i++) {
+				receiveCard(context, player, "minion_wisp");
+			}
+			playCard(context, player, "minion_gnarlroot");
+
+			player.getHand().forEach(c -> print("Card: " + c.getName()));
+
+			player.getMinions().forEach(m -> print("Minion: " + m.getName()));
+
+		});
+
+	}
+
+	@Test
+	public void testFlame() {
+		runGym(((context, player, opponent) -> {
+			receiveCard(context, player, "minion_rebellious_flame");
+			printCards(context, player);
+			playCard(context, player, "minion_wisp");
+			printCards(context, player);
+			playCard(context, player, "spell_arcane_explosion");
+			printCards(context, player);
+
+		}));
+	}
+
+	@Test
+	public void testMeteorStorm() {
```

Comment:
To test cards with random effects, you can either override the random selections or choose a seed with a specific behavior.

To set the seed, `context.setLogic(new GameLogic(10L))` (the first argument to game logic is the seed).

To override a random card selection, you'll have to copy and modify `overrideRandomTarget`, which is pretty involved. It basically uses Mockito to intercept a call to `getRandom` in `GameLogic`, because a random target is ultimately chosen here:

https://github.com/hiddenswitch/Spellsource-Server/blob/ae7215d9da090d501f3b96e461d2b76cab77a875/game/src/main/java/net/demilich/metastone/game/spells/Spell.java#L71